File: dotnet/cafdemalihapa/Files/Executables/NotepadPlusPlus.cs
using Microsoft.Extensions.Logging;

namespace cafdemalihapa.Files.Executables
{
    public class NotepadPlusPlus(
        ILogger<NotepadPlusPlus> logger
    )
    {
        public string GetPath()
        {
            var notePadPlusPlusExecuteFileLocation = CommandLineArgs.GetByKey("--notepaddpp-execute-file-location");
            logger.LogInformation("Note pad plus plus execute file location: {notePadPlusPlusExecuteFileLocation}", notePadPlusPlusExecuteFileLocation);
            return notePadPlusPlusExecuteFileLocation;
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Files/Executables/NotepadPlusPlus.fs
namespace cafdemwimu.console.Files.Executables

open Microsoft.Extensions.Logging
open cafdemwimu.console

type NotepadPlusPlus(logger: ILogger<NotepadPlusPlus>) =
    member _.GetPath() =
        let notePadPlusPlusExecuteFileLocation = CommandLineArgs.GetByKey("--notepaddpp-execute-file-location")
        logger.LogInformation("Note pad plus plus execute file location: {notePadPlusPlusExecuteFileLocation}", notePadPlusPlusExecuteFileLocation)
        notePadPlusPlusExecuteFileLocation
